sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala

```diff
@@ -136,7 +137,8 @@ private[hive] class IsolatedClientLoader(
     (name.startsWith("com.google") && !name.startsWith("com.google.cloud")) ||
     name.startsWith("java.lang.") ||
     name.startsWith("java.net") ||
-    sharedPrefixes.exists(name.startsWith)
+    sharedPrefixes.exists(name.startsWith) ||
+    name.startsWith("[")
```

Comment:
Since it is not super obvious, can we add a comment to say that this is for arrays?